domain/src/main/kotlin/no/nav/su/se/bakover/domain/revurdering/VurderOmBeløpsendringErStørreEnnEllerLik10ProsentAvGjeldendeUtbetaling.kt

```diff
@@ -14,33 +16,44 @@ data class VurderOmBeløpsendringErStørreEnnEllerLik10ProsentAvGjeldendeUtbetal
     val resultat: Boolean
 
     init {
-        fun diffEr10ProsentEllerMer(førsteMånedsbeløp: Int, gjeldendeUtbetalingsbeløp: Int) =
-            abs(førsteMånedsbeløp - gjeldendeUtbetalingsbeløp) >= (0.1 * gjeldendeUtbetalingsbeløp)
-
         val utbetalingstidslinje = TidslinjeForUtbetalinger(
             periode = nyBeregning.periode,
             utbetalingslinjer = eksisterendeUtbetalinger,
         )
 
         val førsteMånedsberegning = nyBeregning.getMånedsberegninger()
             .minByOrNull { it.periode.fraOgMed }!!
-        val gjeldendeUtbetaling = utbetalingstidslinje.gjeldendeForDato(førsteMånedsberegning.periode.fraOgMed)
 
-        resultat = diffEr10ProsentEllerMer(
-            førsteMånedsbeløp = førsteMånedsberegning.finnBeløpFor10ProsentSjekk(),
-            gjeldendeUtbetalingsbeløp = gjeldendeUtbetaling?.beløp ?: 0,
-        )
+        val gjeldendeUtbetaling =
+            utbetalingstidslinje.gjeldendeForDato(førsteMånedsberegning.periode.fraOgMed)?.beløp ?: 0
+
+        val førsteMånedsbeløp = førsteMånedsberegning.finnBeløpFor10ProsentSjekk()
+
+        resultat = when {
+            førsteMånedsbeløp == 0 && gjeldendeUtbetaling == 0 -> {
+                false
+            }
+            førsteMånedsbeløp == 0 && gjeldendeUtbetaling != 0 -> {
+                true
+            }
+            førsteMånedsbeløp != 0 && gjeldendeUtbetaling == 0 -> {
+                true
+            }
+            else -> {
+                abs(førsteMånedsbeløp - gjeldendeUtbetaling) >= (0.1 * gjeldendeUtbetaling)
+            }
+        }
     }
 
     /**
      * Dersom beløpet er så lavt at det havner under minstegrensen må vi sjekke fradragsbeløpet opp mot 10%
      */
     private fun Månedsberegning.finnBeløpFor10ProsentSjekk(): Int {
-        return if (erSumYtelseUnderMinstebeløp()) {
+        return if (getMerknader().contains(Merknad.Beregning.BeløpMellomNullOgToProsentAvHøySats)) {
```

Comment:
Jeg savner en felles plass der vi definerer disse tingene. Jeg likte å kunne spørre MånedBeregning.kt om `erBeløpMellomNullOgToProsentAvHøySats()`. Jeg føler ikke at vi ved å stappe et objekt på en Månedberegning har løst separation og concerns. Dvs. `månedsberegning.erBeløpMellomNullOgToProsentAvHøySats()` vs `beregning.getMerknader().erBeløpMellomNullOgToProsentAvHøySats()`

Reply:
Problemet forut for denne endringen er at førstnenvte `månedsberegning.erBeløpMellomNullOgToProsentAvHøySats()` ikke kan skille på de ulike variantene

a) beløpet var 0 helt fra starten av (for høy inntekt / avslag)
b) beløpet var mellom 0 og 2% av høy, men har blitt "avkortet" til 0 fordi disse pengene ikke skal utbetales (mellom 0 og 2 prosent / avslag)
c) beløpet er under 2% av høy, men det er sosialstønad som har skylden for dette (innvilget)

Tar gjerne i mot en variant som løser det på en bedre måte?